src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -70,13 +74,35 @@ const useDiscovery = () => {
 
   const updateFollowDiscoveryContext = (willFollow, item) => {
     const newFollowedUsers = [...users];
-    const newUser = exhangeFollower(
-      newFollowedUsers,
+    const userId = item.user ? item.user.user_id : item.user_id;
+
+    // Update initial users
+    const newInitialUser = exchangeFollower(newFollowedUsers, willFollow, userId);
+    if (newInitialUser !== null)
+      DiscoveryAction.setDiscoveryInitialUsers(
+        mapUserWith([...discovery?.initialUsers], newInitialUser),
+        discoveryDispatch
+      );
+
+    // Update discovery search users followed section
+    const newSearchFollowedUser = exchangeFollower(discovery?.followedUsers, willFollow, userId);
+    if (newSearchFollowedUser !== null)
+      DiscoveryAction.setNewFollowedUsers(
+        mapUserWith([...discovery?.followedUsers], newSearchFollowedUser),
+        discoveryDispatch
+      );
+
+    // Update discovery search users unfollowed section
+    const newSearchUnfollowedUser = exchangeFollower(
+      discovery?.unfollowedUsers,
       willFollow,
-      item.user ? item.user.user_id : item.user_id
+      userId
     );
-
-    DiscoveryAction.setDiscoveryInitialUsers(mapUser(newUser), discoveryDispatch);
+    if (newSearchUnfollowedUser !== null)
+      DiscoveryAction.setNewUnfollowedUsers(
+        mapUserWith([...discovery?.unfollowedUsers], newSearchUnfollowedUser),
+        discoveryDispatch
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `updateFollowDiscoveryContext` function has been significantly refactored for better state management and data handling. The changes include using the `exchangeFollower` and `mapUserWith` functions to update various sections of the discovery context (initial users, followed users, unfollowed users). This approach is more efficient and maintainable than the previous implementation.



<!-- This is an auto-generated comment by OSS CodeRabbit -->